dts/bindings/mtd/st,m24256.yaml

```diff
@@ -0,0 +1,17 @@
+# SPDX-License-Identifier: Apache-2.0
+# Copyright (c) 2024 Arunmani Alagarsamy
+# Author: Arunmani Alagarsamy <arunmani.a@capgemini.com>
+
+description: st m24256 series I2C EEPROM
+
+compatible: "st,m24256"
```

Comment:
I am guessing this could be a generic st,m24xxx driver, no? i.e. would a m24512 driver really be different?
Interestingly, `boards/shields/x_nucleo_eeprma2` already uses an `st,m24xxx` compatible :)

Reply:
Perhaps this can be made even more generic. From a quick glance, it looks very similar to https://github.com/zephyrproject-rtos/zephyr/pull/69327

Reply:
Just as a hint, M24M02 seems to be different though:
![image](https://github.com/zephyrproject-rtos/zephyr/assets/9367089/10895cb8-fd60-45b0-bddc-df19be744406)

It uses part of the I2C address for the memory address and therefore one I2C slave uses up multiple I2C addresses on the bus.

So probably we shouldn't be too catch-all with the compatible string.